### PR TITLE
Fix warning on MacOS `make check-fuzz` warning on MacOS

### PR DESCRIPTION
**Pull request by citizen-stig**
# Description

Switch to awk for better cross platform support

### CODE REVIEW

**Review by neysofu — COMMENTED** [review]
Am I doing something wrong? The command seems to produce no output on my machine.

```
$ cd fuzz
$ make build-target
cargo +nightly rustc --bin   -- \
	-C debuginfo=full \
	-C debug-assertions \
	-C passes='sancov-module' \
	-C llvm-args='-sanitizer-coverage-level=3' \
	-C llvm-args='-sanitizer-coverage-inline-8bit-counters' \
	-Z sanitizer=address
error: "--bin" takes one argument.
Available binaries:
    accounts_call
    accounts_call_random
    accounts_parse_call_message
    accounts_parse_call_message_random
    address_bech_32_parse_serde
    address_bech_32_try_from_bytes
    bank_call
    bank_parse_call_message
    parse_address
    share_deserialize

make: *** [Makefile:14: build-target] Error 101

# Trying to run awk separately
$ awk -F' = ' '/^\[\[bin\]\]/,/^$$/{ if ($$1 == "name") { gsub(/"/, "", $$2); print $$2 } }' Cargo.toml
$ # No output!
```

**Comment by neysofu** [comment]
BTW I use arch

**Comment by citizen-stig** [comment]
> Am I doing something wrong? The command seems to produce no output on my machine.
> 
> ```
> $ cd fuzz
> $ make build-target
> cargo +nightly rustc --bin   -- \
> 	-C debuginfo=full \
> 	-C debug-assertions \
> 	-C passes='sancov-module' \
> 	-C llvm-args='-sanitizer-coverage-level=3' \
> 	-C llvm-args='-sanitizer-coverage-inline-8bit-counters' \
> 	-Z sanitizer=address
> error: "--bin" takes one argument.
> Available binaries:
>     accounts_call
>     accounts_call_random
>     accounts_parse_call_message
>     accounts_parse_call_message_random
>     address_bech_32_parse_serde
>     address_bech_32_try_from_bytes
>     bank_call
>     bank_parse_call_message
>     parse_address
>     share_deserialize
> 
> make: *** [Makefile:14: build-target] Error 101
> 
> # Trying to run awk separately
> $ awk -F' = ' '/^\[\[bin\]\]/,/^$$/{ if ($$1 == "name") { gsub(/"/, "", $$2); print $$2 } }' Cargo.toml
> $ # No output!
> ```

Thank you for raising that. I was hoping that CI will catch any errors on linux, but it looks like I have to do it myself.

File: fuzz/Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 .PHONY: help
 
-BINARIES := $(shell sed -n '/^\[\[bin\]\]/,/^$$/ { /name\s*=\s*"\(.*\)"/s//\1/p }' Cargo.toml)
+BINARIES := $(shell awk -F' = ' '/^\[\[bin\]\]/,/^$$/{ if ($$1 == "name") { gsub(/"/, "", $$2); print $$2 } }' Cargo.toml)
```
Comment (citizen-stig) [inline fuzz/Makefile]: No accidental merge

Comment (vlopes11) [inline fuzz/Makefile]: nit: we won't catch cases of odd formats such as `name\t= "foo"` or `name="foo"` as we split per ` = ` (i.e. expect a single whitespace)

File: fuzz/Makefile [hunk 2]
```diff
@@ -11,7 +11,7 @@ check: ## Checks that fuzz member compiles
 	cargo check
 
 build-target: ## Build the target fuzz
-	cargo rustc --bin $(TARGET) $(ARGS) -- \
+	cargo +nightly rustc --bin $(TARGET) $(ARGS) -- \
```
Comment (vlopes11) [inline fuzz/Makefile:15]: Any reason why we need to add `nightly` here? Since the root workspace expects `stable`, we might have a divergence there

**Review by vlopes11 — COMMENTED** [review]
As pre-review, I added two questions that might be of relevance here

**Comment by vlopes11** [comment]
@neysofu what is your output for `make targets`?

**Comment by neysofu** [comment]
@vlopes11 that one seems correct:

```
$ make targets
parse_address
address_bech_32_parse_serde
address_bech_32_try_from_bytes
share_deserialize
bank_call
accounts_call
accounts_call_random
bank_parse_call_message
accounts_parse_call_message
accounts_parse_call_message_random
```

**Comment by vlopes11** [comment]
@neysofu I think it works on *nix systems; `build-target` does expect one argument. You can test with `make build` for all targets

**Comment by neysofu** [comment]
Oh yeah, I'm dumb. It works if I set `$TARGET` 😅

**Review by neysofu — APPROVED** [review]
I think @vlopes11 raises a good point about the use of `nightly`. If not strictly needed, I'd remove it. Besides that, LGTM!

File: fuzz/Makefile (hunk 2 above)
Comment (citizen-stig) [reply fuzz/Makefile:15]: I've added nighly because it complained about `-Z sanitizer=address`

Comment (citizen-stig) [reply fuzz/Makefile:15]:
```
TARGET=parse_address make build-target 
cargo rustc --bin parse_address  -- \
                -C debuginfo=full \
                -C debug-assertions \
                -C passes='sancov-module' \
                -C llvm-args='-sanitizer-coverage-level=3' \
                -C llvm-args='-sanitizer-coverage-inline-8bit-counters' \
                -Z sanitizer=address
   Compiling sovereign-sdk-fuzz v0.3.0 (/Users/nikolai/workspace/sovereign/sovereign-sdk/fuzz)
error: the option `Z` is only accepted on the nightly compiler

```

File: fuzz/Makefile (hunk 1 above)
Comment (citizen-stig) [reply fuzz/Makefile]: That's a really good catch! 

I had to torture ChatGPT a little bit more for this one.

**Review by vlopes11 — APPROVED** [review]
LGTM! nit: consider adding a comment on the Makefile on why nightly is needed

**Pull request merged**